tests/core.py

```diff
@@ -386,6 +386,28 @@ def test_bash_operator_multi_byte_output(self):
             output_encoding='utf-8')
         t.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
 
+    def test_bash_operator_kill(self):
+        import subprocess
+        import psutil
+        sleep_time = "100%d" % os.getpid()
+        t = BashOperator(
+            task_id='test_bash_operator_kill',
+            execution_timeout=timedelta(seconds=1),
+            bash_command="/bin/bash -c 'sleep %s'" % sleep_time,
+            dag=self.dag)
+        self.assertRaises(
+            exceptions.AirflowTaskTimeout,
+            t.run,
+            start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
+        sleep(2)
```

Comment:
Putting sleep in a unit test slows down test execution and can have a pretty strongly adverse effect on developers. Can you use mocks or some other mechanism to simulate the sleep or otherwise trigger the timeout prematurely?

Perhaps use a PythonOperator that raises an AirflowTaskTimeout and then verify that on_kill was called?